Tests not passing locally?

When I run `make test` using Python 3.9 and PyTorch 1.9, I currently get:
```
...
  File "/Users/dchiang/Research/diprism/python/_build/pip_packages/lib/python3.9/site-packages/torch_semiring_einsum/equation.py", line 215, in get_bytes_per_element
    return get_bits_per_element(dtype) // 8
  File "/Users/dchiang/Research/diprism/python/_build/pip_packages/lib/python3.9/site-packages/torch_semiring_einsum/equation.py", line 212, in get_bits_per_element
    return torch.iinfo(dtype).bits
TypeError: torch.bool is not supported by torch.iinfo
```

When @grantexley uses Python 3.11.8 and PyTorch 2.4.0, he gets a different error which he will post momentarily.

This is the error:

FAIL: test_infinite_gradient (test.test_sum_product.TestSumProduct.test_infinite_gradient)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/grantmacbook/fggs/test/test_sum_product.py", line 157, in test_infinite_gradient
    self.assertEqual(fgg.factors['fac1'].weights.grad.item(), math.inf)
AssertionError: 0.0 != inf

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,7 @@
 torch>=1.9
+numpy<2
 pydot
-git+https://github.com/bdusell/semiring-einsum
+torch-semiring-einsum>=1.2
 tqdm
 mypy
 packaging

